addons/web/static/tests/components/custom_checkbox_tests.js

```diff
@@ -0,0 +1,49 @@
+odoo.define('web.custom_checkbox_tests', function (require) {
+    "use strict";
+
+    const CustomCheckbox = require('web.CustomCheckbox');
+    const testUtils = require('web.test_utils');
+
+    const { createComponent, dom: testUtilsDom } = testUtils;
+
+    QUnit.module('Components', {}, function () {
+
+        QUnit.module('CustomCheckbox');
+
+        QUnit.test('test checkbox: default values', async function(assert) {
+            assert.expect(5);
+
+            const checkbox = await createComponent(CustomCheckbox, {});
+
+            assert.ok(checkbox.input);
+            assert.notOk(checkbox.input.disabled);
+            assert.ok(checkbox.label);
+            assert.notOk(checkbox.value, 'checkbox should be unchecked');
+
+            testUtilsDom.click(checkbox.label);
```

Comment:
no need to await? Is the DOM updated directly? I mean, is the checkbox already checked here?